django_kafka/consumer.py

```diff
@@ -1,20 +1,43 @@
 import logging
+import traceback
 from pydoc import locate
-from typing import Optional
+from typing import TYPE_CHECKING, Optional
 
 from confluent_kafka import Consumer as ConfluentConsumer
 from confluent_kafka import cimpl
 
 from django_kafka.conf import settings
-from django_kafka.topic import Topic
+from django_kafka.exceptions import DjangoKafkaError
+
+if TYPE_CHECKING:
+    from django_kafka.topic import Topic
 
 logger = logging.getLogger(__name__)
 
 
-class Topics(dict):
-    def __init__(self, *topics: Topic):
-        for topic in topics:
-            self[topic.name] = topic
+class Topics:
```

Comment:
context: `dict` subclass is removed because he topic names can be regex and so there's no help in putting them in a dict-store, instead we need a `get_topic` method which does regex-matching.